manifests/config.pp

```diff
@@ -238,7 +238,12 @@
 
     $job_name = $job_definition['job_name']
 
-    Prometheus::Scrape_job <<| job_name == $job_name |>> {
+    $node_tag = $prometheus::server::collect_tag ? {
+      Undef   => 'prometheus::scrape_job',
```

Comment:
bit of a hacky work around, `tag == Undef` crashes on puppet 5.5 with `Error 500 on SERVER: Server Error: Puppet::Parser::Compiler failed with error NoMethodError: undefined method `downcase' for nil:NilClass`

Reply:
perhaps this should be `undef`, rather than `Undef`

Reply:
cant remember if i tried that, ill give it a go. thanks :)

Reply:
types are checked with `=~`, and a match expression is more lenient whether you're comparing something with a type or a value